torchtext/csrc/regex.cpp

```diff
@@ -6,6 +6,10 @@ Regex::Regex(const std::string& re_str) : re_str_(re_str) {
   compiled_pattern_ = new RE2(re_str_);
 }
 
+Regex::~Regex() {
+  delete compiled_pattern_;
+}
+
```

Comment:
Would this `Regex` object created a lot?
Is `RE* compiled_pattern_` attribute changed during the lifetime of `Regex`?

If the answer is No to both, it seems more semantic to use a smart pointer like `unique_ptr` and get rid of `new ~ delete` idiom entirely.
But if this `Regex` is expected to be instantiated so many times, 
then the overhead might not be negligible.

Reply:
https://github.com/pytorch/text/blob/8c01462995d6c34ff1b123ed6b81a7497e14b96b/torchtext/csrc/gpt2_bpe_tokenizer.cpp#L123 
Yeah if you look at the use case here the `Regex` object is created every time `gpt2_bpe_pre_tokenizer()` function is called. And this function is called for every token in the list of tokens passed into forward. I don't believe it's possible to update the `RE* compiled_pattern_` attribute (or at least the RE2 library doesn't provide any getters or setters to update this attrbiute).

An alternative solution to prevent instantiation of `Regex` object so many times would be for us to create a C++ method that allow us to use the same `Regex` object but change the `compiled_pattern_` by creating a new `RE2` object (which would still require us to do `new` and `delete`).

Let me know your thoughts here and I can do this as a followup.

Reply:
then I think the current implementation should be fine.